mqr/baseline_ussd.py

```diff
@@ -0,0 +1,116 @@
+import logging
+
+import aiohttp
+
+import vaccine.healthcheck_config as config
+from mqr.utils import rapidpro
+from vaccine.base_application import BaseApplication
+from vaccine.states import Choice, ChoiceState, EndState
+from vaccine.utils import (
+    HTTP_EXCEPTIONS,
+    normalise_phonenumber,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_rapidpro():
+    return aiohttp.ClientSession(
+        timeout=aiohttp.ClientTimeout(total=5),
+        headers={
+            "Authorization": f"Token {config.RAPIDPRO_TOKEN}",
+            "Content-Type": "application/json",
+            "User-Agent": "baseline-ussd",
+        },
+    )
+
+
+class Application(BaseApplication):
+    START_SURVEY = "survey_start"
+
+    async def survey_start(self):
+        if config.RAPIDPRO_URL and config.RAPIDPRO_TOKEN:
+            msisdn = normalise_phonenumber(self.inbound.from_addr)
+            urn = ' f"whatsapp:{msisdn.lstrip(' + ')}"'
+
+            for i in range(3):
+                try:
+                    contact = rapidpro.get_contacts(urn=urn).first(
+                        retry_on_rate_exceed=True
+                    )
+
+                    if not contact:
```

Comment:
Happy for this to be sorted in a next PR but this says if the rapidpro contact doesn't exist continue to survey start. We want the contact to exist and we want it to be an MQR contact